### PR TITLE
2019 rav4 adventure pid

### PR DESCRIPTION
**Pull request by pjlao307**
For smoother steering.

Feel free to make these changes manually since my branch has these extraneous commits for some reason.

### CODE REVIEW

File: selfdrive/car/toyota/interface.py [hunk 1]
```diff
@@ -173,9 +173,9 @@ def get_params(candidate, fingerprint, vin="", is_panda_black=False):
       ret.wheelbase = 2.68986
       ret.steerRatio = 14.3
       tire_stiffness_factor = 0.7933
-      ret.lateralTuning.pid.kpV, ret.lateralTuning.pid.kiV = [[0.6], [0.1]]
+      ret.lateralTuning.pid.kpV, ret.lateralTuning.pid.kiV = [[0.15], [0.1]]
```
Comment (rbiasini) [inline selfdrive/car/toyota/interface.py]: seems like ki is too large wrt kp now. What of you use the sienna tuning?
```
ret.lateralTuning.pid.kpV, ret.lateralTuning.pid.kiV = [[0.3], [0.05]]
```
Would like to keep the tuning options limited. Also, anybody with this car that can confirm the tuning is better?

**Comment by ErichMoraga** [comment]
@rbiasini I would take @pjlao307's word for it, and I've worked w/ him a lot for the last 2 years.  He posted several videos including this time lapse...
https://www.youtube.com/watch?time_continue=15&v=tkEvIdzdfUE

Sienna and RAV4 Adventure are very different beasts.  @meanderingthoughts has considered buying one, but I haven't spotted any others in Discord (which I watch very closely).  If @pjlao307's values need to be tweaked later, then so be it.

File: selfdrive/car/toyota/interface.py (hunk 1 above)
Comment (pjlao307) [reply selfdrive/car/toyota/interface.py]: I'll try those values and let you know how it goes.  I think I've tried Kp at 0.05 and it felt the same to me but I didn't look at cabana between the 2 values (0.1 vs 0.05).

**Comment by rbiasini** [comment]
oh, so this tuning would only work for the Rav4 Adventure you think? What about the non-Adventure ones?
In any case, I would be ok with shipping [[0.3],[0.05]] (it's still more than acceptable even for our Corolla) if @pjlao307 can confirm it works well.

**Comment by pjlao307** [comment]
But the TSS2 Corolla has its own block with a set PID value so this change shouldn't affect them, right?
See: https://github.com/commaai/openpilot/blob/devel/selfdrive/car/toyota/interface.py#L182

**Comment by pjlao307** [comment]
I tried Kp=0.3 and Ki=0.05 and it wasn't great.  The steering wheel would still oscillate on curves which became more severe at higher speed and sharper turns.

Here's a comparison between 0.3, 0.05 (left) and my value of 0.15, 0.05 (right) on the same curve going at about the same speed.

Compare

**Comment by pjlao307** [comment]
FWIW I have heard from others with a 2019 Rav4 Hybrid that they don't have this issue but I can't compare cabana drives to see how much different they are from mine.

**Comment by rbiasini** [comment]
About Corolla: yes, it has its own PID tuning. What I mean is that I feel confident with [0.3, 0.05] because another TSS2 car is proven to work well with it.

However, it seems like yours is different from other Rav4 2019. I'm ok differentiating the Adventure trim from the others and assign specific PID tunings, but it seems like we would not be able to distinguish its fingerprint.

**Comment by geohot** [comment]
If this pull request is still relevant, please resubmit it to the master branch. This should allow our development flow to work better, we commit to master, you commit to master, everyone commits to master! Learn more: https://medium.com/@comma_ai/a-2020-theme-externalization-13b33326d8b3

**Pull request closed without merging**